spec/models/assignment_spec.rb

```diff
@@ -90,4 +90,45 @@
       expect(described_class.unsynced_to_mixpanel).to include(mixpanel_failure, mixpanel_nil)
     end
   end
+
+  describe ".for_presentation" do
+    let(:split) { FactoryBot.create(:split) }
+
+    it "doesn't return assignments for finished splits" do
+      finished_split = FactoryBot.create(:split, name: "finished_split", finished_at: Time.zone.now)
+      FactoryBot.create(:assignment, split: finished_split)
+
+      expect(described_class.for_presentation.map(&:split_name)).not_to include("finished_split")
+    end
+
+    it "doesn't return assignments for splits finished before built_at if provided" do
+      finished_split = FactoryBot.create(:split, name: "finished_split", finished_at: Time.zone.now)
+      FactoryBot.create(:assignment, split: finished_split)
+      built_at = 5.minutes.from_now
+
+      expect(described_class.for_presentation(built_at: built_at).map(&:split_name)).not_to include("finished_split")
+    end
+
+    it "returns assignments finished after built_at if provided" do
+      finished_split = FactoryBot.create(:split, name: "finished_split", finished_at: Time.zone.now)
+      FactoryBot.create(:assignment, split: finished_split)
+      built_at = 5.minutes.ago
+
+      expect(described_class.for_presentation(built_at: built_at).map(&:split_name)).to include("finished_split")
+    end
+
+    it "returns nothing if the decision is newer than the assignment" do
+      FactoryBot.create(:assignment, split: split, variant: :hammer_time)
+      split.create_decision!(variant: :touch_this)
+
+      expect(described_class.for_presentation).to be_empty
+    end
+
+    it "returns the assignment if newer than the split decision" do
+      split.create_decision!(variant: :touch_this)
+      FactoryBot.create(:assignment, split: split, variant: :hammer_time)
+
+      expect(described_class.for_presentation.first.variant).to eq("hammer_time")
+    end
+  end
```

Comment:
What does it take for a split to be both decided and finished? Do we care about:

- Assignments older than decision for splits finished after built_at
- Assignments newer than decision for splits finished after built_at
- Assignments newer than decision for splits finished _before_ built_at
- etc?

I know the query construction makes those less likely to be wrong, but I could see someone refactoring it and accidentally missing some kind of interference between the two parts of the query.

Reply:
Yeah, could add those. I don't know if we're gonna get the full cross product though, especially once we add more variables like feature completions and remote kills. Do you have strong feels that any of these are important coverage we're missing?

Reply:
One strategy for testing scopes with many input combinations is to lean more on the factories to set things up in a variety of conditions (like, "i expect the query against [ABCDE] to return [ABD]"). I don't feel super strongly here, but it might be nice to have one additional assignment that falls into two groups just to acknowledge the possibility for future devs.

Reply:
I'm hesitant to add more tests now since the composition of this logic is likely to change in ways that would allow me to get the confidence that things will behave correctly in integration based on unit testing the smaller scopes. I'll take notes on the open Qs from this PR as a tickler file for the overall arc here, though, and sufficient coverage of condition combinations is a thing I'll re-review.